app/javascript/blacklight/search_context.js

```diff
@@ -2,106 +2,157 @@ import Blacklight from './core'
       }
     })
+    // page-links dynamic content depends on a loaded document
+    Blacklight.onLoad(() => {
+      // if page-links container has a data URL, do client-side search context behaviors
+      let clientPageLinks = document.querySelectorAll('.page-links[data-page-links-url]')
+      if (clientPageLinks[0]) {
+        const storedSearch = new URLSearchParams(sessionStorage.getItem("blacklightSearch"))
+        Blacklight.clientAppliedParams(storedSearch)
+        Blacklight.clientPageLinks(clientPageLinks[0].getAttribute('data-page-links-url'), storedSearch)
+      }
+    })
   };
 
   Blacklight.csrfToken = () => document.querySelector('meta[name=csrf-token]')?.content
   Blacklight.csrfParam = () => document.querySelector('meta[name=csrf-param]')?.content
   Blacklight.searchStorage = () => document.querySelector('meta[name=blacklight-search-storage]')?.content
 
-  // this is the Rails.handleMethod with a couple adjustments, described inline:
-  // first, we're attaching this directly to the event handler, so we can check for meta-keys
-  Blacklight.handleSearchContextMethod = function(event) {
-    const link = this
+  /**
+   * for a URL, iterate over searchParams and return a hidden form input for each pair
+   * @param {URL} searchUrl
+   * @returns {string} input element source
+   */
+   const buildInputsFromSearchParams = function(searchUrl) {
+    let inputs = ''
+    for (const [paramName, paramValue] of searchUrl.searchParams.entries()) {
+      inputs += `<input name="${paramName}" value="${paramValue}" type="hidden" />`
+    }
+    return inputs
+  }
 
-    // instead of using the normal href, we need to use the context href instead
-    let contextUrl = new URL(link.getAttribute('data-context-href'), window.location)
-    let target = link.getAttribute('target')
+  /**
+   * build a submittable form to use in the onClick handler of a search result
+   * @param {string} formAction
+   * @param {?string} formTarget
+   * @param {string} formMethod
+   * @param {?string} redirectHref
+   */
+  const buildSearchContextResultForm = function(formAction, formTarget, formMethod, redirectHref) {
+    let actionUrl = new URL(formAction, window.location)
     let csrfToken = Blacklight.csrfToken()
     let csrfParam = Blacklight.csrfParam()
     let form = document.createElement('form')
-    form.action = contextUrl.pathname;
-    const formMethod = link.getAttribute('data-context-method') || 'post'
+    form.action = actionUrl.pathname
     form.method = (formMethod == 'get') ? formMethod : 'post'
 
-    let formContent = ''
-    for (const [paramName, paramValue] of contextUrl.searchParams.entries()) {
-      formContent += `<input name="${paramName}" value="${paramValue}" type="hidden" />`
-    }
+    // check for meta keys.. if set, we should open in a new tab
+    const target = (event.metaKey || event.ctrlKey) ? '_blank' : formTarget
+    if (target) form.target = target
+
+    form.style.display = 'none'
```

Comment:
Is it possible to do `form.hidden = true` here instead? (https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/hidden)

Reply:
yep, done